utils/readJson.py

```diff
@@ -1,6 +1,81 @@
+import csv
 import re
 import requests
 
+from io import StringIO
+
+
+def get_french_address(location, address, postcode, city):
+    full_address = f"{address} {postcode} {city}"
+    url = "https://api-adresse.data.gouv.fr/search"
+    params = {"q": full_address, "limit": 1}
+    response = requests.get(url, params=params)
+    if data:= response.json():
+        return {
+            "location_name": location,
+            "address": data["features"][0]["properties"]["name"],
+            "city": data["features"][0]["properties"]["city"],
+            "departement": data["features"][0]["properties"]["context"].split(',')[0].strip(),
+            "zip_code": data["features"][0]["properties"]["postcode"],
+            "latitude": data["features"][0]["geometry"]["coordinates"][1],
+            "longitude": data["features"][0]["geometry"]["coordinates"][0],
+        }
+    return {
+        "location_name": "",
+        "address": "",
+        "city": "",
+        "department": "",
+        "zip_code": "",
+        "latitude": "",
+        "longitude": "",
+    }
+
+
+def get_default_address(full_address):
+    full_address = full_address.replace(",", " ").replace('  ', ' ').strip()  # for csv
+    words = full_address.split()
+
+    # create a list of address removing one word for each line until three words minimum
+    if lines := "\n".join(' '.join(words[i:]) for i in range(len(words)) if len(words[i:]) >= 3):
+        url = "https://api-adresse.data.gouv.fr/search/csv/"
+        files = {'data': StringIO("query\n" + lines)}
+        response = requests.post(url, files=files)
+        if data_list := list(csv.DictReader(StringIO(response.text))):
+            data_list = list(map(lambda x: {**x, 'result_score': float(x['result_score']) if x['result_score'] != '' else 0.0}, data_list))
+            max_data = max(data_list, key=lambda x:x['result_score'])
+            if max_data["result_score"] > 0.5:  # keep only good results

```

Comment:
I feel that this part should be explained thoroughly, especially given that what we're trying to achieve here is a bit hacky. Commenting the function will do.